go-controller/pkg/ovn/ovn.go

```diff
@@ -609,6 +638,125 @@ func (oc *Controller) GetServiceVIPToName(vip string, protocol kapi.Protocol) (t
 	return namespace, ok
 }
 
+// AddServiceLBToACL associates an empty load balancer with its associated ACL reject rule
+func (oc *Controller) SetServiceACLToLB(lb, vip, acl string) {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	if _, ok := oc.serviceLBMap[lb]; !ok {
+		oc.serviceLBMap[lb] = make(map[string]*LoadBalancerConf)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	if _, ok := oc.serviceLBMap[lb][vip]; !ok {
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	// This should not happen, the entry should be removed for the VIP when the service is deleted for an existing
+	// entry
+	if oc.serviceLBMap[lb][vip] == nil {
+		klog.Errorf("config for existing load balancer is nil. LB: %s, VIP: %s", lb, vip)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	oc.serviceLBMap[lb][vip].RejectACL = acl
+}
+
+// SetServiceEndpointsToLB associates a load balancer with endpoints
+func (oc *Controller) SetServiceEndpointsToLB(lb, vip string, eps []string) {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	if _, ok := oc.serviceLBMap[lb]; !ok {
+		oc.serviceLBMap[lb] = make(map[string]*LoadBalancerConf)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	if _, ok := oc.serviceLBMap[lb][vip]; !ok {
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	// This should not happen, the entry should be removed for the VIP when the service is deleted
+	if oc.serviceLBMap[lb][vip] == nil {
+		klog.Errorf("config for existing load balancer is nil. LB: %s, VIP: %s", lb, vip)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	oc.serviceLBMap[lb][vip].Endpoints = eps
+}
+
+func (oc *Controller) GetServiceLBACL(lb, vip string) (string, bool) {
+	if conf, ok := oc.GetServiceLBConfig(lb, vip); ok {
+		return conf.RejectACL, true
+	}
+	return "", false
+}
+
+func (oc *Controller) GetServiceLBEndpoints(lb, vip string) ([]string, bool) {
+	if conf, ok := oc.GetServiceLBConfig(lb, vip); ok {
+		return conf.Endpoints, true
+	}
+	return []string{}, false
+}
+
+// GetServiceLBConfig retrieves the associated Config for a given load balancer
+func (oc *Controller) GetServiceLBConfig(lb, vip string) (LoadBalancerConf, bool) {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	conf, ok := oc.serviceLBMap[lb][vip]
+	if conf == nil {
+		conf = &LoadBalancerConf{}
+	}
+	return *conf, ok
+}
+
+// GetAllACLsForServiceLB retrieves all of the ACLs for a given load balancer
+func (oc *Controller) GetAllACLsForServiceLB(lb string) []string {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	confMap, ok := oc.serviceLBMap[lb]
+	if !ok {
+		return nil
+	}
+	var acls []string
+	for _, v := range confMap {
+		if v == nil {
```

Comment:
@trozet I'm pretty sure we never add a nil member to the map, right? I'd rather we just panic if we do encounter that, because it means other code is pretty wrong somewhere.

Reply:
yeah I think you are right. There should never be a nil member added.